AI/VCAI/Goals/Explore.cpp

```diff
@@ -179,3 +167,220 @@ bool Explore::fulfillsMe(TSubgoal goal)
 	}
 	return false;
 }
+
+bool Explore::hasReachableNeighbor(const int3 &pos, HeroPtr hero, CCallback * cbp, VCAI * vcai) const
+{
+	for(crint3 dir : int3::getDirs())
+	{
+		int3 tile = pos + dir;
+		if(cbp->isInTheMap(tile))
+		{
+			auto paths = vcai->ah->getPathsToTile(hero, tile);
+
+			return paths.size();
+		}
+	}
+
+	return false;
+}
+
+int Explore::howManyTilesWillBeDiscovered(
+	const int3 & pos, 
+	int radious, 
+	CCallback * cbp, 
+	HeroPtr hero, 
+	std::function<bool (const int3 &)> filter) const
```

Comment:
Moved and added a filter. explorationNewPoint does not use the filter anymore. explorationBestNeighbor uses hasVisibleNeighbors as the filter